packages/storage/src/cookies.ts

```diff
@@ -93,20 +100,22 @@ export const cookieStorage: StorageWithOptions<CookieOptions> = addClearMethod({
         return
       }
       const pageEvent: PageEvent = options?.getRequest?.() || useRequest();
-      if (!("responseHeaders" in pageEvent)) // Check if we really got a pageEvent
+      if (!pageEvent.responseHeaders) // Check if we really got a pageEvent
         return
       const responseHeaders = pageEvent.responseHeaders as Headers;
-      const cookies = responseHeaders.get("Set-Cookie")?.split(",").filter((cookie) => deserializeCookieOptions(cookie, key) == null) ?? [];
+      const cookies = responseHeaders.get("Set-Cookie")?.split(",")
+        .filter((cookie) => !cookie.match(`\\s*${key}\\s*=`)) ?? [];
```

Comment:
thanks, just copied from the other two matches, they have to be because they use a variable in the regex. I know that's not very elegant, because someone could use special characters for the name and then everything would break down, but that wasn't my solution initially, so not my fault i guess :)

Reply:
if you have an idea how to fix that without impacting performance, please tell me. My only idea was to url encode the key, but than some developers might wonder why the key is different on the browser than in the code.